test/DynamoCoreTests/Models/DynamoModelEventsTest.cs

```diff
@@ -0,0 +1,509 @@
+
+            //Assert
+            DynCmd.RequestDispatcherBeginInvoke -= DynamoModel_RequestDispatcherBeginInvoke;
+            CurrentDynamoModel.EvaluationCompleted -= CurrentDynamoModel_EvaluationCompleted;
+            //This will validate that the local handler was executed and set the flag in true
+            Assert.IsTrue(dispatcherExecuted);
+            Assert.IsTrue(evaluationCompleted);
+        }
+
+       
+
+        /// <summary>
+        /// This test method will execute the event OnRequestDispatcherBeginInvoke
+        /// </summary>
+        [Test]
+        [Category("UnitTests")]
+        public void TestOnRequestMigrationStatusDialog()
+        {
+            //Arrange
+            //This will subscribe our local method to the RequestMigrationStatusDialog event
+            DynCmd.RequestMigrationStatusDialog += DynamoModel_RequestMigrationStatusDialog;
+
+            //Act
+            //This will execute the OnRequestMigrationStatusDialog() in the DynamoModelEvents class
+            DynCmd.OnRequestMigrationStatusDialog(new SettingsMigrationEventArgs(
+                SettingsMigrationEventArgs.EventStatusType.Begin));
+
+            //Assert
+            DynCmd.RequestMigrationStatusDialog -= DynamoModel_RequestMigrationStatusDialog;
+            //This will validate that the local handler was executed and set the flag in true
+            Assert.IsTrue(migrationStatusDialog);
+        }
+
+        /// <summary>
+        /// This test method will execute the event OnRequestLayoutUpdate
+        /// </summary>
+        [Test]
+        [Category("UnitTests")]
+        public void TestOnRequestLayoutUpdate()
+        {
+            //Arrange
+            EvaluationCompletedEventArgs e = new EvaluationCompletedEventArgs(true, new Exception("Test"));
+
+            //Act
+            //This will subscribe our local method to the RequestLayoutUpdate event
+            CurrentDynamoModel.RequestLayoutUpdate += Model_RequestLayoutUpdate;
+
+            //This method needs to be called directly since we are not loading any dyn file to manage events
+            CurrentDynamoModel.OnRequestLayoutUpdate(this, e);
+
+            //Assert
+            CurrentDynamoModel.RequestLayoutUpdate -= Model_RequestLayoutUpdate;
+            //This will validate that the local handler was executed and set the flag in true
+            Assert.IsTrue(layoutUpdate);
+        }
+
+        /// <summary>
+        /// This test method will execute the event OnWorkspaceClearing
+        /// </summary>
+        [Test]
+        [Category("UnitTests")]
+        public void TestOnWorkspaceClearing()
+        {
+            //Arrange
+            //This will subscribe our local method to the RequestLayoutUpdate event
+            CurrentDynamoModel.WorkspaceClearing += CurrentDynamoModel_WorkspaceClearing;
+
+            //Act
+            //Internall this will execute the OnWorkspaceClearing() method in DynamoModelEvents
```

Comment:
Spelling

Reply:
@QilongTang 
I fixed the spelling in two test methods in the next commit:
https://github.com/DynamoDS/Dynamo/pull/10612/commits/22270346ba76b9fcdb4cd08be0be75ce65a1f326